### PR TITLE
MWPW-130556 | Adding WebPage Schema

### PR DESCRIPTION
**Pull request by apganapa**
This will add webPage schema block to pages. Currently block created needs to be added on the document so that webPage gets added.
this is for testing IN locale with 5 pages.
Once SEO team confirms to go forward with all locales and all pages then the code will get moved to scripts.js

Fix MWPW-130556

Test URLs:
- Before: https://main--express-website--adobe.hlx.page/in/express/feature/image/test-schema-qr-code-generator
- After: https://mwpw-130556--express-website--adobe.hlx.page/in/express/feature/image/test-schema-qr-code-generator

### CODE REVIEW

**Review by JingleH — APPROVED** [review]
Looks good and I am approving, but I think we need to be very cautious before bringing this into scripts.js and potentially affecting our performance. We can talk about how to proceed later

File: express/blocks/schemas/schemas.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      webApplicationUrl = cellValue;
+    }
+  });
+  const webPageSchemaScript = document.createElement('script');
+  webPageSchemaScript.setAttribute('type', 'application/ld+json');
+  const webSchemaJson = {
+    '@context': 'https://schema.org',
+    '@type': 'WebPage',
+    url: window.location.href,
+    '@id': ''.concat(window.location.href, '/#webpage'),
```
Comment (qiyundai) [inline express/blocks/schemas/schemas.js:37]: nit: I think interpolation would be nicer here (and all other places where we used ''.concat) -- `${window.location.href}/#webpage`;

Comment (qiyundai) [reply express/blocks/schemas/schemas.js:37]: GH removed my backticks... '``'

File: express/blocks/schemas/schemas.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import {
+  getLocale,
+} from '../../scripts/scripts.js';
+
+function decorateSchemasBlocks($block) {
```
Comment (qiyundai) [inline express/blocks/schemas/schemas.js:16]: Since it's a new block, let's start dropping the '$' 😄

File: express/blocks/schemas/schemas.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  const rows = Array.from($block.children);
+  const locale = getLocale(window.location);
+  const urlPrefix = locale === 'us' ? '' : `/${locale}`;
+  const homePageLocaleUrl = ''.concat('https://www.adobe.com/', urlPrefix, '/');
+  let webApplicationUrl;
+  rows.forEach(($row) => {
+    const cells = Array.from($row.children);
+    const cellValue = cells[0].innerText;
+    if (cellValue === '') {
+      webApplicationUrl = document.querySelectorAll('p.button-container')[0].getElementsByTagName('a')[0].href;
```
Comment (qiyundai) [inline express/blocks/schemas/schemas.js:26]: querySelectorAll('tag.class')[0] could be refactored into querySelector('tag.class');

Comment (qiyundai) [reply express/blocks/schemas/schemas.js:26]: 2 more issues I see here:
- This line of chained queries can throw a typeError without `?` before the `.`.
- Even with `?`, are we OK with the `about > url` in the schema being 'undefined'

**Review by qiyundai — APPROVED** [review]
lgtm!

**Pull request merged**